eng/linkerTests/linkerCorrectnessTests.targets

```diff
@@ -0,0 +1,109 @@
+<Project>
+  <PropertyGroup>
+    <LinkerTestDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'linkerTests'))</LinkerTestDir>
+    <LinkerTestProjectsDir>$([MSBuild]::NormalizeDirectory('$(LinkerTestDir)', 'projects'))</LinkerTestProjectsDir>
+    <TestDotNetPath>$([MSBuild]::NormalizePath('$(DotNetRoot)', 'dotnet'))</TestDotNetPath>
+    <ProjectTemplate>$([MSBuild]::NormalizePath('$(MSBuildThisFileDirectory)', 'project.csproj.template'))</ProjectTemplate>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <TestConsoleAppSourceFiles Include="$(MSBuildProjectDirectory)/*.cs" />
```

Comment:
Can we instead rely on `EnableDefaultItems` and the `Compile` item? Makes more sense to me instead of manually globbing the source files. cc @ercistj for opinion.

Reply:
The actual projects are not compiling these *.cs files, so they are not added to the project so Compile item is empty. These source files are copied into a new generated project and that is where we compile them.